packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -987,155 +956,24 @@ export const Table = <
               >
                 <Box {...headerGroupProps}>
                   {headerGroup.headers.map((column, index) => {
-                    const { onClick, ...restSortProps } =
-                      column.getSortByToggleProps();
-
-                    const columnHasExpanders =
-                      hasAnySubRows &&
-                      index ===
-                        headerGroup.headers.findIndex(
-                          (c) => c.id !== SELECTION_CELL_ID, // first non-selection column is the expander column
-                        );
-
-                    if ([undefined, 0].includes(column.minWidth)) {
-                      // override "undefined" or zero min-width with default value
-                      column.minWidth = columnHasExpanders
-                        ? COLUMN_MIN_WIDTHS.withExpander
-                        : COLUMN_MIN_WIDTHS.default;
-
-                      // set the minWidth to the user provided width instead if the width is less than the default minWidth
-                      if (
-                        typeof column.width === 'number' &&
-                        column.minWidth > column.width
-                      ) {
-                        column.minWidth = column.width;
-                      }
-                    }
-
-                    const columnProps = column.getHeaderProps({
-                      ...restSortProps,
-                      className: cx(
-                        'iui-table-cell',
-                        {
-                          'iui-actionable': column.canSort,
-                          'iui-sorted': column.isSorted,
-                          'iui-table-cell-sticky': !!column.sticky,
-                        },
-                        column.columnClassName,
-                      ),
-                      style: {
-                        ...getCellStyle(column, !!state.isTableResizing),
-                        ...(columnHasExpanders && getSubRowStyle({ density })),
-                        ...getStickyStyle(column, visibleColumns),
-                        flexWrap: 'wrap',
-                        columnGap: 'var(--iui-size-xs)',
-                      },
-                    });
-
                     return (
-                      <Box
-                        {...columnProps}
+                      <ColumnHeader<T>
                         {...column.getDragAndDropProps()}
-                        key={columnProps.key}
-                        title={undefined}
-                        ref={(el) => {
-                          columnResizeRef(el, column);
-                        }}
```

Comment:
That didn't work because the component is being instantiated inside of a map function, so it isn't at the top level of the component. [discussion](https://github.com/iTwin/iTwinUI/pull/2092#discussion_r1672610769).